src/test/java/com/example/rets_api/repository/BasicPropertyWithOneEntityRepositoryJPATest.java

```diff
@@ -0,0 +1,240 @@
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getAnimalPolicy());
+        assertTrue(baseTests.compareAnimalPolicy(propertySaved.getAnimalPolicy(), propertyEntityToCompare.getAnimalPolicy()));
+        assertNotNull(propertySaved.getAnimalPolicy().getAnimalPolicyId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndLotData() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFields();
+        propertyToSave.setLotData(createLotDataEntity());
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithAllNestedFields();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getLotData());
+        assertTrue(baseTests.compareLotData(propertySaved.getLotData(), propertyEntityToCompare.getLotData()));
+        assertNotNull(propertySaved.getLotData().getLotDataId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndContactInformation() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFields();
+        propertyToSave.setContactInformation(createContactInformationEntity());
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithAllNestedFields();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getContactInformation());
+        assertNotNull(propertySaved.getContactInformation().getContactInformationId());
+        assertTrue(baseTests.compareContactInformation(propertySaved.getContactInformation(), propertyEntityToCompare.getContactInformation()));
+        assertNotNull(propertySaved.getContactInformation().getContactInformationId());
+    }
+
+    @Test
+    public void should_store_a_PropertyAndListingPrice() {
+        //Creating
+        PropertyEntity propertyToSave = EntityUtilsTest.createPropertyEntityWithBasicFields();
+        propertyToSave.setListingPrice(createListingPriceEntity());
+        PropertyEntity propertyEntityToCompare = EntityUtilsTest.createPropertyEntityWithAllNestedFields();
+
+        //Saving
+        PropertyEntity propertySaved = propertyRepository.saveAndFlush(propertyToSave);
+
+        //Validating
+        checkAllBasicFieldsFromProperty(propertySaved);
+
+        assertNotNull(propertySaved.getListingPrice());
+        assertTrue(baseTests.compareListingPrice(propertySaved.getListingPrice(), propertyEntityToCompare.getListingPrice()));
+        assertNotNull(propertySaved.getListingPrice().getListingPriceId());
+    }
+
+    protected void checkAllBasicFieldsFromProperty(PropertyEntity property){
```

Comment:
You can use the method from BasicPropertyRepositoryJPATest

Reply:
done